addons/native_dialog/osx_dialog.m

```diff
@@ -185,7 +185,8 @@ int _al_show_native_message_box(ALLEGRO_DISPLAY *display,
       NSArray* buttons = [button_text componentsSeparatedByString: @"|"];
       [box setMessageText:[NSString stringWithUTF8String: al_cstr(fd->title)]];
       [box setInformativeText:[NSString stringWithUTF8String: al_cstr(fd->mb_text)]];
-      [box setAlertStyle: NSWarningAlertStyle];
+      [box setAlertStyle: NSAlertStyleWarning];
```

Comment:
drive by fix of deprecated constants

Reply:
Note this is for 10.12 and later. What's the earliest version that can be used to build Allegro currently?
https://developer.apple.com/documentation/appkit/nsalertstyle/nsalertstylewarning

Reply:
README_osx mentions versions earlier than 10.12, so I'll just revert this part.

Reviewer note: I used the github web editor to make the change, so you'll want to squash on merge.